scripts/deleteExpiredAccessTokens.js

```diff
@@ -0,0 +1,354 @@
+ * @return {Array} - The access tokens.
+ */
+gpii.accessTokens.filterExpiredAccessTokens = function (responseString, options) {
+    if (options.deleteAll) {
+        fluid.log("Deleting all access tokens...");
+    } else {
+        fluid.log("Filtering for expired access tokens...");
+    }
+    var tokens = JSON.parse(responseString);
+    var expiredTokens = [];
+    options.totalTokens = 0;
+    if (tokens.rows) {
+        fluid.each(tokens.rows, function (aRow) {
+            var aToken = aRow.value.authorization;
+            if (options.deleteAll || Date.now() > Date.parse(aToken.timestampExpires)) {
+                expiredTokens.push(aToken);
+            }
+            options.totalTokens++;
+        });
+        options.accessTokens = expiredTokens;
+    }
+    return expiredTokens;
+};
+
+/**
+ * Given the expired access tokens, mark them for deletion.
+ * @param {Array} tokens - Array of access tokens to delete from the database.
+ * @return {Array} - the access tokens marked for deletion.
+ */
+gpii.accessTokens.markAccessTokensForDeletion = function (tokens) {
+    fluid.each(tokens, function (aToken) {
+        aToken._deleted = true;
+    });
+    return tokens;
+};
+
+/**
+ * Create an http request for a bulk docs POST request using the given data.
+ * @param {Object} dataToPost - JSON data to POST and process in bulk.
+ * @param {ResponseCallback} responseHandler - http response callback for the
+ *                                             request.
+ * @param {Object} options - Post request options:
+ * @param {PostRequestOptions} options.postOptions - the POST request specifics.
+ * @return {http.ClientRequest} - An http request object.
+ */
+gpii.accessTokens.createPostRequest = function (dataToPost, responseHandler, options) {
+    var batchPostData = JSON.stringify({"docs": dataToPost});
+    options.postOptions.headers["Content-Length"] = Buffer.byteLength(batchPostData);
+    var batchDocsRequest = http.request(options.postOptions, responseHandler);
+    batchDocsRequest.write(batchPostData);
+    return batchDocsRequest;
+};
+
+/**
+ * Generate a response handler, setting up the given promise to resolve/reject
+ * at the correct time.
+ * @param {ResponseDataHandler} handleEnd - Function that processes the response
+ *                                          data when the response receives an
+ *                                          "end" event.
+ * @param {Object} options - Data loader options passed to `handleEnd()`.
+ * @param {Promise} promise - Promise to resolve/reject on a response "end" or
+ *                           "error" event.
+ * @param {String} errorMsg - Optional error message to prepend to the error
+ *                            received from a response "error" event.
+ * @return {ResponseCallback} - Reponse callback function suitable for an http
+ *                              request.
+ */
+gpii.accessTokens.createResponseHandler = function (handleEnd, options, promise, errorMsg) {
```

Comment:
It seems there are some functions shared between this script and scripts/deleteAndLoadSnapsets.js. This function is one of them. Can you please move shared utility functions into a separate file in the /shared sub-directory to reduce the duplicates. Thanks.

Reply:
Sure.